### PR TITLE
test: [TERR-425] React Component Code Coverage and Error Validation

### PR DESCRIPTION
**Pull request by sanchezelton**
This PR adds code coverage reporting (using React Testing Library's use of Jest and Istanbul) to React component testing in the Network Map Grafana Plugin.

This also implements error validation to the UI for the CustomTextArea component and refactors its to utilize a controlled component implementation.

## Checklist

-   [x] PR Title format: `type: [TERR-XXX] Short description`
    -   `type` is one of: 'build', 'ci', 'chore', 'docs', 'feat', 'fix', 'perf', 'refactor', 'revert', 'style', 'test'
    -   `TERR-XXX` refers to a Jira issue.
    -   If this is a breaking change prefix the description with "BREAKING CHANGE:"
-   [x] There is a related Jira issue for this pull request
-   [x] Description should be a meaningful summary of thea changes you are proposing
-   [x] For breaking changes prefix the title with "BREAKING CHANGE:" and include details in the description
-   [x] This PR is as small as possible
-   [x] This PR includes tests related to these changes, or existing tests provide coverage
-   [x] This PR has updated documentation as appropriate

### CODE REVIEW

File: src/components/CustomTextArea.tsx
Comment (jkafader-esnet) [inline src/components/CustomTextArea.tsx:1]: This file probably needs a more fine-toothed comb. Let's go over this one interactively

File: src/components/lib/esmap.js
Comment (jkafader-esnet) [inline src/components/lib/esmap.js:1]: Why delete these?

File: src/types.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export interface LayerOptions {
 }
 
 export interface MapOptions {
-  topologySource: string;
+  useConfigurationUrl: boolean;
```
Comment (jkafader-esnet) [inline src/types.ts:27]: topologySource is the source for this information now. Delete

File: tsconfig.json
Comment (jkafader-esnet) [inline tsconfig.json:1]: Unsure why tsconfig changes were linked to this PR

File: tsconfig.json [hunk 2]
```diff
@@ -1,14 +1,16 @@
 {
   "extends": "./.config/tsconfig.json",
-  "include": ["src", "types"],
```
Comment (jkafader-esnet) [inline tsconfig.json:3]: almost certainly we don't want to delete this file

File: yarn.lock
Comment (jkafader-esnet) [inline yarn.lock:1]: Probably want to pull this from upstream unchanged, almost certainly we don't want to alter yarn.lock in this case -- the dependencies are carefully hand managed due to requested changes from Grafana.

File: .eslintrc
Comment (jkafader-esnet) [inline .eslintrc:1]: Where did these changes come from?

File: .gitignore [hunk 3]
```diff
@@ -42,6 +42,9 @@ e2e-results/
 .DS_store
 *.code-workspace
 
+# vscode
+*.code-workspace
```
[the remark this answers is not in the document]
Comment (sanchezelton) [reply .gitignore]: Removed.

File: src/types.ts (hunk 1 above)
Comment (sanchezelton) [reply src/types.ts]: Removed.

File: tsconfig.json (hunk 2 above)
Comment (sanchezelton) [reply tsconfig.json:3]: I believe the settings are set in the parent tsconfig, but it is harmless to have a more direct reference here, adding it back in.

File: package.json [hunk 4]
```diff
@@ -80,8 +89,8 @@
   "dependencies": {
     "@emotion/css": "11.10.6",
     "d3": "^7.0.0",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "react": "17.0.1",
+    "react-dom": "17.0.1",
```
[the remark this answers is not in the document]
Comment (sanchezelton) [reply package.json]: Reverted to 18.2.0

File: package.json [hunk 5]
```diff
@@ -30,16 +31,23 @@
...
     "@swc/core": "^1.3.90",
     "@swc/helpers": "^0.5.0",
     "@swc/jest": "^0.2.26",
+    "@testing-library/dom": "^10.4.0",
+    "@testing-library/jest-dom": "^6.5.0",
+    "@testing-library/react": "12.1.2",
+    "@testing-library/user-event": "^14.5.2",
     "@types/jest": "^29.5.0",
-    "@types/lodash": "^4.14.194",
+    "@types/lodash": "4.14.187",
```
[the remark this answers is not in the document]
Comment (sanchezelton) [reply package.json]: Reverted to ^4.14.194

File: package.json [hunk 6]
```diff
@@ -80,8 +89,8 @@
   "dependencies": {
     "@emotion/css": "11.10.6",
     "d3": "^7.0.0",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "react": "17.0.1",
```
[the remark this answers is not in the document]
Comment (sanchezelton) [reply package.json]: Reverted to 18.2.0

File: yarn.lock
Comment (sanchezelton) [reply yarn.lock:1]: Reverted to revision from `end-to-end-tests`

File: tsconfig.json (hunk 2 above)
Comment (sanchezelton) [reply tsconfig.json:3]: Restored to include it.

File: src/components/lib/esmap.js
Comment (sanchezelton) [reply src/components/lib/esmap.js:1]: `text` is not a valid attribute for outputted path elements for SVG, is not retained by attempting to set it and is unreliable for use as a key finding edges.

File: src/components/CustomTextArea.tsx [hunk 7]
```diff
@@ -1,46 +1,231 @@
...
-  suffix?: ReactNode;
-}
-
 function unescape(str) {
   return String(str)
     .replace(/&amp;/g, '&')
     .replace(/&lt;/g, '<')
     .replace(/&gt;/g, '>')
-    .replace(/&quot;/g, '"');
+    .replace(/&quot;/g, "'");
```
[the remark this answers is not in the document]
Comment (sanchezelton) [reply src/components/CustomTextArea.tsx]: Updated.

**Pull request merged**